lib/em-hiredis/req_resp_connection.rb

```diff
@@ -0,0 +1,77 @@
+
+module EventMachine::Hiredis
+  module ReqRespConnection
+    include EventMachine::Hiredis::EventEmitter
+
+    def initialize
+      super
+      @response_queue = []
+      @connected = false
+    end
+
+    def send_command(df, command, args)
+      @response_queue.push(df)
+      puts "send #{command} #{args}"
+      send_data(marshal(command, *args))
+      return df
+    end
+
+    # EM::Connection callback
+    def connection_completed
+      puts "Connection completed"
+      @reader = ::Hiredis::Reader.new
+      @connected = true
+      emit(:connected)
+    end
+
+    # EM::Connection callback
+    def receive_data(data)
+      @reader.feed(data)
+      until (reply = @reader.gets) == false
+        puts "reply #{reply}"
+        df = @response_queue.shift
+        if df
+          if RuntimeError === reply
+            e = EM::Hiredis::RedisError.new(reply.message)
+            e.redis_error = reply
+            df.fail(e)
+          else
+            df.succeed(reply)
+          end
+        else
+          emit(:replies_out_of_sync)
+          close_connection
+        end
+      end
+    end
+
+    # EM::Connection callback
+    def unbind
+      puts "Unbind"
+      @response_queue.each { |df| df.fail(EM::Hiredis::Error.new('Redis connection lost')) }
+      @response_queue.clear
+      if @connected
+        emit(:disconnected)
+      else
+        emit(:connection_failed)
+      end
+    end
+
+    protected
+
+    COMMAND_DELIMITER = "\r\n"
+
+    def marshal(*args)
```

Comment:
The implementation is equivalent, and arguably cleaner, but removing the splat does complicate things for the invoker, which currently does `marshal(command, *args)`

"I want a variable number of arguments given to me as an iterable" seems a reasonable use of a splat to me, what was it that annoyed you about it?